hnswlib/hnswalg.h

```diff
@@ -271,7 +271,7 @@ namespace hnswlib {
 
                 std::pair<dist_t, tableint> current_node_pair = candidate_set.top();
 
-                if ((-current_node_pair.first) > lowerBound) {
+                if ((-current_node_pair.first) > lowerBound && top_candidates.size() == ef) {
```

Comment:
I think it should be 
```python
if ((-current_node_pair.first) > lowerBound && (top_candidates.size() == ef || has_deletions == false))
```

Reply:
I added it :) Thank you for your kind explanation.